[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only} failing

**Build Scans:**
- [elasticsearch-intake #43832 / 9.3.8_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vs5gtuydgkk7i)
- [elasticsearch-intake #43826 / 9.3.8_bwc-snapshots](https://gradle-enterprise.elastic.co/s/6vrtnws3uhlcw)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.3.8#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only}" -Dtests.seed=F9F67702D1EB8FEB -Dtests.bwc=true -Dtests.locale=st-LS -Dtests.timezone=Brazil/DeNoronha -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dsearch.vectors%2F41_knn_search_byte_quantized_bfloat16%2FKNN%20Vector%20similarity%20search%20only%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: field [hits.hits] doesn't have length [1]
Expected: <1>
     but: was <2>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only}
- [main] 1 consecutive failures in step 9.3.8_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only} (1.2% fail rate in 172 executions)
- [main] 2 failures in step 9.3.8_bwc-snapshots (15.4% fail rate in 13 executions)
- [main] 2 failures in pipeline elasticsearch-intake (15.4% fail rate in 13 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -637,9 +637,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecSpatialShapesIT
   method: test {csv-spec:spatial_shapes.convertFromStringParseError}
   issue: https://github.com/elastic/elasticsearch/issues/154157
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/KNN Vector similarity search only}
-  issue: https://github.com/elastic/elasticsearch/issues/154197
 - class: org.elasticsearch.xpack.esql.action.FromDatasetIT
   method: testWildcardSpanningIndexAndDatasetSucceeds
   issue: https://github.com/elastic/elasticsearch/issues/154200

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_byte_quantized_bfloat16.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_byte_quantized_bfloat16.yml
@@ -202,6 +202,9 @@ setup:
 
 ---
 "KNN Vector similarity search only":
+  - requires:
+      cluster_features: [ "lucene_10_4_upgrade" ]
+      reason: "Approximate int8_hnsw scoring differs between Lucene 10.3 and 10.4+ quantizers"
   - do:
       search:
         index: hnsw_byte_quantized
@@ -220,6 +223,9 @@ setup:
   - match: {hits.hits.0.fields.name.0: "moose.jpg"}
 ---
 "Vector similarity with filter only":
+  - requires:
+      cluster_features: [ "lucene_10_4_upgrade" ]
+      reason: "Approximate int8_hnsw scoring differs between Lucene 10.3 and 10.4+ quantizers"
   - do:
       search:
         index: hnsw_byte_quantized

